packages/core/addon/chart-builders/apex.js

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2020, Yahoo Holdings Inc.
+ * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
+ */
+
+/**
+ * @typedef {object} requestType
+ * @property {array} dimensions - the dimensions of the data
+ * @property {array} intervals - the time interval of the data
+ * @property {object} logicalTable - the granularity of the time intervals of the data
+ * @property {array} metrics - the metrics of the data
+ */
+
+/**
+ * @typedef {object} returnType
+ * @property {array} labels - the labels to be displayed on the chart (if applicable)
+ * @property {array} series - the metric series containing data
+ */
+
+/**
+ * shapes data into apex-friendly format
+ * @param {requestType} request
+ * @param {array} rows
+ * @returns {returnType}
+ */
+export function shapeData(request, rows) {
```

Comment:
What is the scope of this function? General normalization for apex or visualization type specific?

Reply:
Should we call this `normalize`?